lib/internal/readline/interface.js

```diff
@@ -1323,42 +1322,72 @@ class Interface extends InterfaceConstructor {
    * }} InterfaceAsyncIterator
    * @returns {InterfaceAsyncIterator}
    */
-  [SymbolAsyncIterator]() {
-    if (this[kLineObjectStream] === undefined) {
-      if (Readable === undefined) {
-        Readable = require('stream').Readable;
+  async *[SymbolAsyncIterator]() {
+    const queue = new FixedQueue();
+    const hwm = 128;
+    const lwm = hwm / 2;
+
+    let resume = null;
+    let error = null;
+    let errored = false;
+    let len = 0;
+
+    const errorHandler = (err) => {
+      error = err;
+      errored = true;
+
+      resume?.();
+      resume = null;
+    };
+
+    const lineHandler = (line) => {
+      queue.push(line);
+
+      len += 1;
```

Comment:
isn't `len` just `queue.length` ?

Reply:
(or is that not exposed on a FixedQueue?)

Reply:
FixedQueue doesn't expose it